include/seqan3/search/search.hpp

```diff
@@ -197,7 +209,20 @@ inline auto search(queries_t && queries,
 }
 
 //!\cond DEV
-//! \overload
+// Overload for a single query (not a collection of queries)
+//!\overload
+template <fm_index_specialisation index_t,
+          typename query_t,
+          typename configuration_t = decltype(search_cfg::default_configuration)>
+    requires !std::ranges::forward_range<std::ranges::range_value_t<query_t>>
```

Comment:
Ok, this is currently only working because the above function is not constrained. Which is also not right. As soon as the above function requires forward range on the queries it breaks, because of ambigious overloads. 
I propose the following change to also reflect the actual requirements on the types:
```cpp
template <std::ranges::forward_range queries_t>
    requires std::ranges::random_access_range<std::ranges::range_value_t<queries_t>>
void foo(queries_t && qry)
{
    std::cout << "Take range of ranges!\n";
}

template <std::ranges::random_access_range query_t>
    requires (!std::ranges::range<std::ranges::range_value_t<query_t>>)
void foo(query_t && qry)
{
    std::cout << "Take single range!\n";
}
```

Reply:
I would not include the requirements of the random access range here. We decided to only require stuff that is needed for overload resolution and static assert the other requirements (as is done in `validate_query`)

Reply:
> I would not include the requirements of the random access range here. We decided to only require stuff that is needed for overload resolution and static assert the other requirements (as is done in validate_query)

0. Obviously, that is needed for overload resoultion.
1. That was a wrong decision and is outdated or should be if not done so yet. 
2. Validating the query is not top-level and in detail but it should cause the error on the top-level. It would be different if there were a configuration dependent assert, so that a certain property is only available for a certain configuration. Here you would throw an exception, but since you can infer it at compile time you would prefer the static_assert. But here we are talking about primary types independent of the configuration, so I think it does not apply to this.
3. What happened with your opinion that it is better to see the type requirements immediately? Seemed to be relevant in a private member function of a detail class. 😝 
4. Because with gcc-10 you get really good diagnostics anyway. Don't we have a range_of_ranges concept also? If not might be worth to discuss.

Reply:
0. Why? the dimension is needed fot overload resolution not the random access restriction?
1. I disagree because you can add a rational in the static_assert comment.
2.  I don't understand what you are trying to tell me. `validate_query_type()` has only static_asserts, no throw statements and is evaluated top-level in the generic function `search()` not within the algorithm. I consider this top-level, even if it passes through a view::single wrapper beforehand.
3. out of context
4. Having great stuff in GCC10 does not mean we should abandon nice error messages in the former versions.